File: apps/customer_app/src/api/menu.ts
import { requestJson } from './client';

export type LiveMerchantMenuItem = {
  id: number;
  merchant_id: number;
  name: string;
  price_sos: number;
  is_active: boolean;
};

export async function listMerchantMenu(merchantEmail: string) {
  return requestJson<LiveMerchantMenuItem[]>({
    path: `/api/merchant/menu?merchant_email=${encodeURIComponent(merchantEmail)}`,
  });
}

File: backend/app/models/menu.py
from datetime import datetime

from sqlalchemy import Boolean, DateTime, ForeignKey, Integer, String
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.db import Base


class MerchantMenuItem(Base):
    __tablename__ = "merchant_menu_items"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    merchant_id: Mapped[int] = mapped_column(ForeignKey("users.id"), nullable=False, index=True)
    name: Mapped[str] = mapped_column(String(140), nullable=False, index=True)
    price_sos: Mapped[int] = mapped_column(Integer, nullable=False)
    is_active: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False, index=True)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)
    updated_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    merchant = relationship("User", foreign_keys=[merchant_id])

    @property
    def merchant_email(self) -> str | None:
        return self.merchant.email if self.merchant else None
